R/transactions.R

```diff
@@ -43,4 +57,29 @@ setGeneric("dbCommit",
 setGeneric("dbRollback",
   def = function(conn, ...) standardGeneric("dbRollback"),
   valueClass = "logical"
-)
\ No newline at end of file
+)
+
+#' @aliases withTransaction,DBIConnection-method
+#' @export
+#' @rdname transactions
+setGeneric("withTransaction",
+  def = function(conn, statement, ...) standardGeneric("withTransaction"),
+  valueClass = "logical"
+)
+
+#' @export
+setMethod("withTransaction", "DBIConnection", function(conn, statement, ...) {
```

Comment:
I don't see why we need to reevaluate multiple times in withTransaction(). A function is definitely the right thing for a callback, as in dbGetQueryChunked() -- although I have yet to understand the use of the callback here.